src/unix/process.c

```diff
@@ -451,15 +452,25 @@ int uv_spawn(uv_loop_t* loop,
     r = read(signal_pipe[0], &exec_errorno, sizeof(exec_errorno));
   while (r == -1 && errno == EINTR);
 
+#define _UV_WAIT() \
+  { \
+    do \
+      err = waitpid(pid, &status, 0); \
+    while (err == -1 && errno == EINTR); \
+    assert(err == pid); \
+  }
```

Comment:
I'd rather you'd not use a macro for this.  Avoiding duplication is a good thing but this obscures it too much.